File: common/src/main/java/com/rabtman/common/base/BaseApplication.java
package com.rabtman.common.base;

import android.app.Application;
import android.content.Context;
import android.support.multidex.MultiDex;
import com.rabtman.common.utils.SystemUtils;


public abstract class BaseApplication extends Application {

  private CommonApplicationLike mCommonApplicationLike;


  @Override
  public void onCreate() {
    super.onCreate();
    this.mCommonApplicationLike = new CommonApplicationLike(this);
    this.mCommonApplicationLike.onCreate();

    boolean defaultProcess = SystemUtils.getCurProcessName().equals(getPackageName());
    if (defaultProcess) {
      this.mCommonApplicationLike.onDefaultProcessCreate();
      this.onDefaultProcessCreate();
    }
  }

  @Override
  protected void attachBaseContext(Context base) {
    super.attachBaseContext(base);
    MultiDex.install(this);
  }

  /**
   * 程序终止的时候执行
   */
  @Override
  public void onTerminate() {
    super.onTerminate();
    this.mCommonApplicationLike.onTerminate();
  }

  /**
   * 在默认进程中进行操作
   */
  public void onDefaultProcessCreate() {

  }

}

File: common/src/main/java/com/rabtman/common/base/BaseApplication.kt
package com.rabtman.common.base

import android.app.Application
import android.content.Context
import androidx.multidex.MultiDex
import com.rabtman.common.utils.SystemUtils

abstract class BaseApplication : Application() {
    private var mCommonApplicationLike: CommonApplicationLike? = null
    override fun onCreate() {
        super.onCreate()
        mCommonApplicationLike = CommonApplicationLike(this)
        mCommonApplicationLike!!.onCreate()
        val defaultProcess = SystemUtils.getCurProcessName() == packageName
        if (defaultProcess) {
            mCommonApplicationLike!!.onDefaultProcessCreate()
            onDefaultProcessCreate()
        }
    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        MultiDex.install(this)
    }

    /**
     * 程序终止的时候执行
     */
    override fun onTerminate() {
        super.onTerminate()
        mCommonApplicationLike!!.onTerminate()
    }

    /**
     * 在默认进程中进行操作
     */
    open fun onDefaultProcessCreate() {}
}